fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/detail/LaoDetailActivity.java

```diff
@@ -60,8 +60,11 @@ protected void onCreate(@Nullable Bundle savedInstanceState) {
     setContentView(R.layout.lao_detail_activity);
     mViewModel = obtainViewModel(this);
     mViewModel.subscribeToLao(
-        (String) Objects.requireNonNull(getIntent().getExtras()).get("LAO_ID"));
-    if (getIntent().getExtras().get("FRAGMENT_TO_OPEN").equals("LaoDetail")) {
+        (String) Objects.requireNonNull(getIntent().getExtras()).get(getString(R.string.lao_id)));
```

Comment:
The `string.xml` should be used for text that will be displayed to the user, and you should keep using constants defined in the code for things like keys and so.

https://stackoverflow.com/questions/16160047/android-strings-xml-vs-static-constants